src/python/tensorflow_cloud/core/containerize.py

```diff
@@ -157,32 +157,20 @@ def _create_docker_file(self):
                 if float(v[0] + "." + v[1]) <= 2.1:
                     self.docker_base_image += "-py3"
 
-        if not self._base_image_exist():
+        # TF nightly image names have the substring "dev"
+        if "dev" in self.docker_base_image:
             warnings.warn(
                 "Docker base image {} does not exist.".format(
-                    self.docker_base_image))
-            if "dev" in self.docker_base_image:
-                # Except for the latest TF nightly, other nightlies
-                # do not have corresponding docker images.
-                newtag = "nightly"
-                if self.docker_base_image.endswith("-gpu"):
-                    newtag += "-gpu"
-                self.docker_base_image = (
-                    self.docker_base_image.split(":")[0] + ":" + newtag
-                )
-                warnings.warn("Using the latest TF nightly build.")
-            else:
-                warnings.warn(
-                    "Using the latest stable TF docker image available: "
-                    "`tensorflow/tensorflow:latest`"
-                    "Please see "
-                    "https://hub.docker.com/r/tensorflow/tensorflow/ "
-                    "for details on available docker images."
-                )
-                newtag = "tensorflow/tensorflow:latest"
-                if self.docker_base_image.endswith("-gpu"):
-                    newtag += "-gpu"
-                self.docker_base_image = newtag
+                self.docker_base_image))
+            # Except for the latest TF nightly, other nightlies
```

Comment:
This logic appears to make assumptions on TensorFlow docker image's tag scheme.

Could you drop a comment about what exactly the assumption on the docker image that this program is making on [`tensorflow/tensorflow`](https://hub.docker.com/r/tensorflow/tensorflow/tags?page=1) docker container's versioning and tagging scheme?

It seems to me that if `tensorflow/tensorflow:latest-devel-gpu` is asked, it rewrites to `tensorflow:tensorflow:nightly-gpu`.   Customer's custom docker image might also be following this tagging conversion that might include `-devel-`, which may also cause this logic to kick-in, but `nightly` might not exist.

Reply:
Improved the string check. "devel" image names will not be included in this anymore.